### PR TITLE
Implement oval parsing for debian and ubuntu

### PR DESCRIPTION
**Pull request by ldelossa**
This PR implements correct OVAL parsing for debian and ubuntu moving
away from incorrect regex parsing of the oval data.

goval-parser is bumped to obtain the necessary datastructures.

Signed-off-by: ldelossa <ldelossa@redhat.com>

### CODE REVIEW

File: ubuntu/parser.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+func (u *Updater) Parse(ctx context.Context, r io.ReadCloser) ([]*claircore.Vulnerability, error) {
+	log := zerolog.Ctx(ctx).With().
+		Str("component", "debian/Updater.Parse").
+		Logger()
+	ctx = log.WithContext(ctx)
+	log.Info().Msg("starting parse")
+	defer r.Close()
+	root := oval.Root{}
+	if err := xml.NewDecoder(r).Decode(&root); err != nil {
+		return nil, fmt.Errorf("debian: unable to decode OVAL document: %w", err)
```
Comment (hdonnay) [inline ubuntu/parser.go:24]:
```suggestion
		Str("component", "ubuntu/Updater.Parse").
		Logger()
	ctx = log.WithContext(ctx)
	log.Info().Msg("starting parse")
	defer r.Close()
	root := oval.Root{}
	if err := xml.NewDecoder(r).Decode(&root); err != nil {
		return nil, fmt.Errorf("ubuntu: unable to decode OVAL document: %w", err)
```

File: pkg/ovalutil/dpkg.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+				// that object, so just skip tests with a single object reference
+				// and no associated state object.
+				continue
+			}
+			if len(test.ObjectRefs) != len(test.StateRefs) {
+				log.Debug().Str("test_ref", criterion.TestRef).Msg("object refs and state refs are not in pairs. moving to next criterion")
+				continue
+			}
+			// look at each object,state pair the test references
+			// and create a vuln if an evr tag if found
```
Comment (hdonnay) [inline pkg/ovalutil/dpkg.go:56]:
```suggestion
			// and create a vuln if an evr tag is found
```

This typo probably also exists on the RPM version :(

File: pkg/ovalutil/dpkg.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+					vuln := *protoVuln
+					vuln.FixedInVersion = state.EVR.Body
+					name := object.Name
+
+					// if the dpkginfo_object>name field has a var_ref it indicates
+					// a variable lookup for all packages affected by this vuln is necessary.
+					//
+					// if the name.Ref field is empty it indicates a single package is affected
+					// by the vuln and that package's name is in name.Body.
+					if len(name.Ref) > 0 {
```
Comment (hdonnay) [inline pkg/ovalutil/dpkg.go:88]: This would be easier to read if written into early return (continue, in this case) form instead of if-else

**Review by hdonnay — DISMISSED** [review]
Some typos in log/error strings, but code looks good.

File: pkg/ovalutil/dpkg.go (hunk 2 above)
Comment (ldelossa) [reply pkg/ovalutil/dpkg.go]: fixed in both.

**Review by hdonnay — DISMISSED** [review]
LGTM

**Pull request merged**